e2e/test/scenarios/question-reproductions/reproductions-3.cy.spec.js

```diff
@@ -2610,3 +2614,55 @@ describe("issue 32499", () => {
     }
   });
 });
+
+describe("issue 23449", () => {
+  beforeEach(() => {
+    H.restore();
+    cy.signInAsAdmin();
+  });
+
+  it("Remapped fields should work in a model", () => {
+    // set the metadata for review
+    cy.visit(
+      `/admin/datamodel/database/${SAMPLE_DB_ID}/schema/${SAMPLE_DB_SCHEMA_ID}/table/${SAMPLE_DATABASE.REVIEWS_ID}`,
+    );
+
+    cy.findByTestId("column-RATING").findByLabelText("Field settings").click();
+    cy.findAllByTestId("select-button").contains("Use original value").click();
+    cy.findByLabelText("Custom mapping").click();
+
+    const values = { 1: "A", 2: "B", 3: "C", 4: "D", 5: "E" };
+    for (const [key, value] of Object.entries(values)) {
+      cy.findByDisplayValue(key).clear().type(value);
+    }
+    cy.button("Save").click();
+
+    // make a model on Reviews
+    cy.findByRole("link", { name: "Exit admin" }).click();
+    H.navigationSidebar().findByLabelText("Browse models").click();
+    cy.findByLabelText("Create a new model").click();
+    cy.findByRole("link", { name: /Use the notebook editor/ }).click();
+    H.entityPickerModal().findByText("Reviews").click();
+
+    // Remove Body column
+    cy.findByLabelText("Pick columns").click();
+    H.popover().findByText("Body").click();
+
+    // save model
+    cy.button("Save").click();
+    cy.findByTestId("save-question-button").click();
+
+    // check that the data renders
+    const checkTable = () => {
```

Comment:
I would pull the function definition out of the test and into the describe block.